mihir_task 1 - instantiated instruction decoder in top.v

MM871 opened the pull request.

MM871 committed:
task 1 - instantiated instruction decoder in top.v

joon2022park commented:
Instantiate 3 modules

(Use Boris's top.v as a baseline/reference)

- ID stage (Instruction Decode)
- MA stage (Memory access stage)

MM871 committed:
Update top.v

instantiated MA and ID

MM871 committed:
muxes made and instantiated

instantiated in bottom of top.v and created the module in a seperate file Mux.v

MM871 committed:
Tried case statements for muxes, feedback needed

TheDeepestSpace committed:
Merge remote-tracking branch 'upstream/main' into 11-integrate-instruction-decode-with-memory-access

TheDeepestSpace commented:
hey @MM871, I just merged master into you branch and resolved all conflicts; i think we are missing a couple of variables there (see this job output -- https://github.com/UTOSS/risc-v/actions/runs/16403567149/job/46346493415?pr=25)

 great job on using the case statements, you nailed it!

TheDeepestSpace committed:
connect mux outputs to alu

TheDeepestSpace committed:
fix mux conditions

TheDeepestSpace committed:
add missing registers

TheDeepestSpace committed:
temporarily wire rs1 and 2 to alu directly

TheDeepestSpace commented:
I just threw in the missing registers and it all compiles now! Thanks @MM871, gonna merge this now
